Assets/MixedRealityToolkit/_Core/Interfaces/InputSystem/IMixedRealityInputSource.cs

```diff
@@ -1,32 +1,78 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-using System.Collections;
 using Microsoft.MixedReality.Toolkit.Internal.Definitions;
+using System.Collections;
+using System.Collections.Generic;
 
 namespace Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem
 {
     /// <summary>
-    /// Interface for an input source.
-    /// An input source can be any user defined action that generally comes from a physical controller, sensor, or device.
+    /// Interface for a Mixed Reality Toolkit input source.
+    /// An input source can be any user defined input that generally comes from a physical controller, sensor, or device.
     /// </summary>
     public interface IMixedRealityInputSource : IEqualityComparer
     {
+        /// <summary>
+        /// The ID assigned to the Input Source
+        /// </summary>
         uint SourceId { get; }
 
+        /// <summary>
+        /// The Name assigned to the Input Source
+        /// </summary>
         string SourceName { get; }
 
+        /// <summary>
+        /// Outputs the current state of the Input Source, whether it is tracked or not. As defined by the SDK / Unity.
+        /// </summary>
+        InputSourceState InputSourceState { get; }
+
+        /// <summary>
+        /// The designated hand that the Input Source is managing, as defined by the SDK / Unity.
+        /// </summary>
+        Handedness Handedness { get; }
+
+        /// <summary>
+        /// The list of Pointers attached to the Input Source
+        /// </summary>
         IMixedRealityPointer[] Pointers { get; }
 
         /// <summary>
-        /// Returns the input info that the input source can provide.
+        /// Details the list of available buttons / interactions available from the Input Source.
         /// </summary>
-        InputType[] Capabilities { get; }
+        Dictionary<InputType,InteractionDefinition> Interactions { get; }
```

Comment:
As much as I like this new patter, I feel like it's a bit redundant to have the input type both as a key and in the interaction definition itself.

Reply:
There is that thought, but it happens when it's working in both roles. Will see if I can remove the duplication